cate/core/opimpl.py

```diff
@@ -327,6 +316,26 @@ def normalize_missing_time(ds: xr.Dataset) -> xr.Dataset:
             time_coverage_end = pd.to_datetime(time_coverage_end)
         except BaseException:
             pass
+    if time_coverage_start or time_coverage_end:
+        return time_coverage_start, time_coverage_end
+    filename = ds.encoding.get('source', '')
+    return get_timestamps_from_string(filename)
+
+
+def normalize_missing_time(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Add a time coordinate variable and their associated bounds coordinate variables
```

Comment:
It now covers the case, that the time is only given in the file name, could you add this in the docstring?